crates/bevy_ecs/src/component.rs

```diff
@@ -283,22 +284,48 @@ impl ComponentInfo {
 /// one `World` to access the metadata of a `Component` in a different `World` is undefined behavior
 /// and must not be attempted.
 #[derive(Debug, Copy, Clone, Hash, Ord, PartialOrd, Eq, PartialEq)]
-pub struct ComponentId(usize);
+pub struct ComponentId {
+    index: u32,
+    target: u32,
+}
 
 impl ComponentId {
+    const NO_TARGET: u32 = u32::MAX;
+
     /// Creates a new [`ComponentId`].
     ///
     /// The `index` is a unique value associated with each type of component in a given world.
     /// Usually, this value is taken from a counter incremented for each type of component registered with the world.
     #[inline]
     pub const fn new(index: usize) -> ComponentId {
-        ComponentId(index)
+        ComponentId {
+            index: index as u32,
+            target: Self::NO_TARGET,
+        }
+    }
+
+    /// Creates a new relation [`ComponentId`].
+    ///
+    /// The relation is of the input `ComponentId` and the target is the given entity.
+    ///
+    /// ## Examples
+    /// ```rust,ignore
+    /// let child_of: ComponentId = todo!(); // Get the component id of `ChildOf`
+    /// let parent: Entity = todo!(); // Get some parent
+    /// let child_of_parent: ComponentId = child_of.relation(parent);
+    /// ```
+    #[inline]
+    pub const fn relation(self, target: Entity) -> Self {
+        ComponentId {
+            index: self.index,
+            target: target.index(),
+        }
     }
 
     /// Returns the index of the current component.
     #[inline]
-    pub fn index(self) -> usize {
-        self.0
```

Comment:
safe because of https://github.com/bevyengine/bevy/blob/cd0a642fa19476dd9876aa71198ee28b62e0e3f2/crates/bevy_ecs/src/lib.rs#L6-L7

Reply:
I'm not sure what this means? Are you asking me to put a justification comment for the `as` cast?

Reply:
nothing, just a general comment during review 🙂